File: Behavioral/Strategy/Python/bitcoin_payment.py
# bitcoin_payment.py

from payment_strategy import PaymentStrategy

class BitcoinPayment(PaymentStrategy):
  def __init__(self, wallet_address):
    self.wallet_address = wallet_address

  def pay(self, amount):
    print(f"Paid ${amount} using Bitcoin (Wallet: {self.wallet_address})")

File: Behavioral/Strategy/JavaScript/bitcoinPayment.js
// bitcoinPayment.js

const PaymentStrategy = require("./paymentStrategy");

class BitcoinPayment extends PaymentStrategy {
  constructor(walletAddress) {
    super();
    this.walletAddress = walletAddress;
  }

  pay(amount) {
    console.log(`Paid $${amount} using Bitcoin (Wallet: ${this.walletAddress})`);
  }
}

module.exports = BitcoinPayment;
